downloader_cli/download.py

```diff
@@ -165,16 +165,23 @@ def _parse_URL(self):
         """
         if match(r"^https?://*", self.URL):
             return [self.URL]
-
+       
         rel_path = path.expanduser(self.URL)
+        if rel_path.startswith('file://'):
+            # the URL begins with a file protocol
+            rel_path = rel_path[len('file://'):]
+
+        # get the realpath of the URL
+        # follows linux `realpath`
+        rel_path = path.realpath(rel_path)
+
         # Put a check to see if the file is present
         if not path.exists(rel_path) or not path.isfile(rel_path):
             print("{}: not a valid name or is a directory".format(rel_path))
             exit(-1)
 
-        # If it's not an URL, read the contents.
-        with open(rel_path, "r") as RSTREAM:
-            return RSTREAM.read().split("\n")
+        # If it's not an URL, provide the file:// protocol URL.
```

Comment:
@srevinsaju Doesn't this break the ability to download files that are passed in a file seperated by a newline??

Reply:
@deepjyoti30 Is it possible to download multiple URL separated by a newline?

Reply:
If that feature is supported by normal URLs,  like `https` or `http`, then that would be supported by `file://` protocol too, I have used the exact same logic you implemented for `https` protocol for the `file://` protocol

Reply:
I haven't had time to read it yet, but couldn't these be chained in the read loop?

Reply:
What do you mean?

Reply:
Currently, line 177 above reads the file, splits it on newlines, and returns the list of presumably-URLs.

So don't return that list. First, go through and find the URLs you want to alter, prepend `file://`, and then return the modified list.

If you wanna make it *really* robust, try the URL first, see what's there, and *then* pick the protocol, but I'm sure not about to put my money where my mouth is! You might =P

Reply:
@ChanceNCounter Yeah I think that sounds better. If it is a `file://` or an URL we can directly get to downloading.

However, if it is a file, we need to extract all the URL's present in it, be it `file://` or `https://` or `http://`.

@srevinsaju I tried your commits, it is surely breaking the ability to pass a file which contains URL's seperated by newlines in order to support bulk downloading.

Reply:
Thanks for testing this PR. But, I guess I do not understand the logic implemented, I will open an issue instead :D